Lib/pathlib.py

```diff
@@ -604,14 +575,19 @@ def parent(self):
         drv = self.drive
         root = self.root
         parts = self._parts
-        if len(parts) == 1 and (drv or root):
+        if len(parts) == bool(drv or root):
```

Comment:
This feels like a change in logic. Is it intentional? I.e. If len(parts) == 0 and not drv and not root -- we return self

Reply:
Yeah, it's intentional. The idea of this `if:` block is to `return self` is we're at the top level, as an optimization, because:

```python
Path('/').parent == Path('/')
Path('.').parent == Path('.')
```

Previously it was working for the first case but not the second.

Reply:
To be clear, this method would still work if the `if:` block was taken out